htsget-search/src/storage/local.rs

```diff
@@ -139,7 +140,70 @@ mod tests {
     });
   }
 
-  // TODO add tests for `LocalStorage::url`
+  #[test]
+  fn url_of_non_existing_key() {
+    with_local_storage(|storage| {
+      let result = storage.url("non-existing-key", UrlOptions::default());
+      assert_eq!(
+        result,
+        Err(StorageError::InvalidKey("non-existing-key".to_string()))
+      );
+    });
+  }
+
+  #[test]
+  fn url_of_folder() {
+    with_local_storage(|storage| {
+      let result = storage.url("folder", UrlOptions::default());
+      assert_eq!(result, Err(StorageError::NotFound("folder".to_string())));
+    });
+  }
+
+  #[test]
+  fn url_with_forbidden_path() {
+    with_local_storage(|storage| {
+      let result = storage.url("folder/../../passwords", UrlOptions::default());
+      assert_eq!(
+        result,
+        Err(StorageError::InvalidKey(
+          "folder/../../passwords".to_string()
+        ))
+      );
+    });
+  }
+
+  #[test]
+  fn url_of_existing_key() {
+    with_local_storage(|storage| {
+      let result = storage.url("folder/../key1", UrlOptions::default());
+      let expected = Url::new(format!(
+        "file://{}",
+        storage.base_path().join("key1").to_string_lossy()
+      ))
+      .with_headers(Headers::default().with_header("Range", "bytes=-"));
```

Comment:
if this is happening, I'd say that this is a bug in the implementation. It should detect that there is no range defined and skip that header.

Reply:
[Open-ended ranges are supported in curl](https://everything.curl.dev/libcurl-http/ranges) and generally in [http ranges RFC 7233](https://tools.ietf.org/html/rfc7233)...

A single `-` could mean "all bytes", but since it's not reflected in the IETF RFC linked above, I'd say that it's invalid... @CastilloDel I'd say that you can fix it in the context of this PR since it's on topic and it shouldn't be such a big changeset I reckon.

Reply:
... or actually, looking a bit closer, as a separate PR. You are both right, should be separate.

Reply:
So this PR should be safe to merge now, @chris-zen?... provided that @CastilloDel opens a corresponding issue for what is discussed here so that we don't forget? ;)

Reply:
Yes please, after creating the issue we can merge this PR. Thanks @CastilloDel !

Reply:
Sorry, I 've been so busy these days! I'll create the issue now and hopefully I will be able to create a PR for it soon!